tests/acceptance/editor-undo-redo-test.js

```diff
@@ -99,6 +92,46 @@ test('undo/redo the insertion of multiple characters', (assert) => {
   });
 });
 
+test('adding and deleting characters break the undo group/run', (assert) => {
```

Comment:
Adding a test for the timeout functionality would also be good here (and is another reason that it could/should be configurable — at least for tests).